experimental/examples/codeviewer/shared/src/jvmMain/kotlin/org/jetbrains/codeviewer/platform/JvmFile.kt

```diff
@@ -58,22 +64,37 @@ fun java.io.File.toProjectFile(): File = object : File {
             override val size get() = size
 
             override fun get(index: Int): String {
-                val startPosition = lineStartPositions[index]
-                val length = if (index + 1 < size) lineStartPositions[index + 1] - startPosition else
-                    byteBufferSize - startPosition
-                // Only JDK since 13 has slice() method we need, so do ugly for now.
-                byteBuffer.position(startPosition)
-                val slice = byteBuffer.slice()
-                slice.limit(length)
+                val position = lineRange(index)
+                val slice = byteBuffer.slice(position.first, position.last - position.first)
                 return StandardCharsets.UTF_8.decode(slice).toString()
             }
+
+            private fun lineRange(index: Int): IntRange {
+                val startPosition = lineStartPositions[index]
+                val nextLineIndex = index + 1
+                var endPosition = if (nextLineIndex < size) lineStartPositions[nextLineIndex] else byteBufferSize
+
+                // Remove line endings from the range
+                while (endPosition > startPosition) {
+                    val lastSymbol = byteBuffer[endPosition - 1]
+                    when (lastSymbol.toInt().toChar()) {
+                        '\n', '\r' -> endPosition--
+                        else -> break
+                    }
+                }
+                return startPosition..endPosition
+            }
         }
     }
 }
 
-private fun java.io.File.readLinePositions(
-    starts: IntList
-) {
+// Backport slice from JDK 13
+private fun ByteBuffer.slice(index: Int, length: Int): ByteBuffer {
+    position(index)
+    return slice().limit(length) as ByteBuffer
```

Comment:
limit(length) returns Buffer and not ByteBuffer on Android

Reply:
Good catch. Thanks for the fix!